core/common/src/main/kotlin/com/twitter/rules/core/util/ASTNodes.kt

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 Twitter, Inc.
+// SPDX-License-Identifier: Apache-2.0
+package com.twitter.rules.core.util
+
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.PsiComment
+import org.jetbrains.kotlin.lexer.KtTokens
+
+fun ASTNode.lastChildLeafOrSelf(): ASTNode {
```

Comment:
this is 1:1 from the ktlint helpers, needed to copy it here to avoid bringing the ktlint dependency over to the detekt modules.